misc/ls-extensions/modules/project-design-service/src/main/java/io/ballerina/projectdesign/generators/GeneratorUtils.java

```diff
@@ -0,0 +1,88 @@
+import io.ballerina.compiler.syntax.tree.SyntaxKind;
+import io.ballerina.projectdesign.model.ModelElementLocation;
+import io.ballerina.projectdesign.model.service.ServiceAnnotation;
+import io.ballerina.tools.text.LineRange;
+
+import static io.ballerina.projectdesign.ProjectDesignConstants.DISPLAY_ANNOTATION;
+import static io.ballerina.projectdesign.ProjectDesignConstants.ID;
+import static io.ballerina.projectdesign.ProjectDesignConstants.LABEL;
+
+/**
+ * Provide utils functions for component model generating.
+ *
+ * @since 2201.2.2
+ */
+public class GeneratorUtils {
+
+    public static ModelElementLocation getElementLocation(String filePath, LineRange lineRange) {
+
+        ModelElementLocation.LinePosition startPosition = ModelElementLocation.LinePosition.from(
+                lineRange.startLine().line(), lineRange.startLine().offset());
+        ModelElementLocation.LinePosition endLinePosition = ModelElementLocation.LinePosition.from(
+                lineRange.endLine().line(), lineRange.endLine().offset()
+        );
+        return ModelElementLocation.from(filePath, startPosition, endLinePosition);
+
+    }
+
+    public static ServiceAnnotation getServiceAnnotation(NodeList<AnnotationNode> annotationNodes, String filePath) {
+
+        String id = "";
+        String label = "";
+        ModelElementLocation elementLocation = null;
+        for (AnnotationNode annotationNode : annotationNodes) {
+            String annotationName = annotationNode.annotReference().toString().trim();
+            if (annotationName.equals(DISPLAY_ANNOTATION)) {
+                if (annotationNode.annotValue().isPresent()) {
+                    SeparatedNodeList<MappingFieldNode> fields = annotationNode.annotValue().get().fields();
+                    elementLocation = getElementLocation(filePath, annotationNode.lineRange());
+                    for (MappingFieldNode mappingFieldNode : fields) {
+                        if (mappingFieldNode.kind() == SyntaxKind.SPECIFIC_FIELD) {
+                            SpecificFieldNode specificFieldNode = (SpecificFieldNode) mappingFieldNode;
+                            String name = specificFieldNode.fieldName().toString().trim();
+                            if (specificFieldNode.valueExpr().isPresent()) {
+                                ExpressionNode expressionNode = specificFieldNode.valueExpr().get();
+                                String expressionNodeStr = expressionNode.toString().trim();
+                                String annotation = expressionNodeStr.replace("\"", "");
+                                if (name.equals(ID)) {
+                                    id = annotation;
+                                } else if (name.equals(LABEL)) {
+                                    label = annotation;
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            break;
```

Comment:
Shall we try to reduce the cognitive complexity (i.e. the number of “if” conditions, nested loops) of this util method?

Reply:
Add `continue` in some levels. Please review

Reply:
Now it looks cleaner :)